crates/core/src/dht/dht_reducers.rs

```diff
@@ -158,15 +158,18 @@ pub fn reduce_queue_holding_workflow(
 ) -> Option<DhtStore> {
     let action = action_wrapper.action();
     let (pending, maybe_delay) = unwrap_to!(action => Action::QueueHoldingWorkflow);
+
+    // TODO: TRACING: this is where we would include a Span, so that we can resume
+    // the trace when the workflow gets popped (see instance.rs), but we can't do that
+    // until we stop cloning the State, because Spans are not Cloneable.
```

Comment:
I'm not sure if we ever will stop cloning the state. As a matter of fact, we recently added more State cloning (instead of tossing around lock guards which lead to deadlocks) and have made that cloning cheaper since the State is comprised almost only of `Arc`s and `im` collections..

Could spans live in `Arc`s inside the state?